torch/csrc/jit/runtime/register_distributed_ops.cpp

```diff
@@ -110,6 +117,28 @@ RegisterOperators reg_rpc_ops(
            return 0;
          },
          aliasAnalysisFromSchema()),
+     Operator(
+         "aten::dist_backward(int context_id, Tensor[] roots, bool retain_graph=False) -> ()",
+         [](Stack& stack) {
+           RECORD_FUNCTION(
+               kDistAutogradBackwardProfilingKey, std::vector<c10::IValue>());
+           bool retain_graph = pop(stack).toBool();
+           auto roots = pop(stack).toTensorList();
+           int64_t context_id = pop(stack).toInt();
+           torch::autograd::variable_list variables;
+           for (const auto& root : roots) {
+             variables.emplace_back(root);
+           }
+           try {
+             dist_autograd::DistEngine::getInstance().execute(
+                 context_id, variables, retain_graph);
+           } catch (python_error& e) {
+             // FIXME: crashes if exception type is not RuntimeError
+             throw std::runtime_error(e.what());
+           }
+           return 0;
+         },
+         aliasAnalysisConservative()),
```

Comment:
Will this be a concern that we have duplicate code for pybind and builtin op? 
Can defining it as a custom op helps since it can make single copy of C++ code be called from both python and torchscript?

Reply:
exposed the C++ API instead in https://github.com/pytorch/pytorch/pull/38656, so we share impls between both